src/main/scala/mesosphere/marathon/state/TaskRepository.scala

```diff
@@ -0,0 +1,41 @@
+package mesosphere.marathon.state
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.metrics.Metrics
+import mesosphere.marathon.tasks.TaskIdUtil
+import mesosphere.util.ThreadPoolContext.context
+
+import scala.concurrent.Future
+
+class TaskRepository(
+  protected val store: EntityStore[MarathonTaskState],
+  protected val metrics: Metrics)
+    extends EntityRepository[MarathonTaskState] {
+
+  val maxVersions = None
+
+  def task(key: String): Future[Option[MarathonTask]] = currentVersion(key).map {
+    case Some(taskState) => Some(taskState.toProto)
+    case _               => None
+  }
+
+  def task(appId: PathId, taskId: String): Future[Option[MarathonTask]] =
+    task(TaskIdUtil.getKey(appId, taskId))
+
+  def tasksKeys(appId: PathId): Future[Iterable[String]] = {
+    allIds().map(_.filter(name => name.startsWith(appId.safePath + ID_DELIMITER)))
+  }
+
+  def store(appId: PathId, task: MarathonTask): Future[MarathonTask] = {
+    val key: String = TaskIdUtil.getKey(appId, task.getId)
+    this.store.store(key, MarathonTaskState(task)).map(_.toProto)
+  }
+
+  // TODO: this is needed because EntitRepo does not allow colons as separators and won't return any keys in allIds
```

Comment:
Changed that. Migration is provided – if this will not go into 0.12 I'd update the names …